core/services/pipeline/runner.go

```diff
@@ -312,14 +378,69 @@ func logTaskRunToPrometheus(trr TaskRunResult, spec Spec) {
 func (r *runner) ExecuteAndInsertFinishedRun(ctx context.Context, spec Spec, vars Vars, l logger.Logger, saveSuccessfulTaskRuns bool) (runID int64, finalResult FinalResult, err error) {
```

Comment:
Yeah I'm not sure what the behavior should be on the old method. Since it's used by code that always expects a response I was considering reverting the changes (back to normal insert) to it and instead returning an error if it's called with a job spec containing `async`.